cmd/kube-controller-manager/app/core.go

```diff
@@ -434,7 +435,7 @@ func startGarbageCollectorController(ctx ControllerContext) (http.Handler, bool,
 		ignoredResources[schema.GroupResource{Group: r.Group, Resource: r.Resource}] = struct{}{}
 	}
 	garbageCollector, err := garbagecollector.NewGarbageCollector(
-		dynamicClient,
+		metadataClient,
 		ctx.RESTMapper,
 		deletableResources,
 		ignoredResources,
```

Comment:
this is still passing in `ctx.GenericInformerFactory`, which is still using `dynamicInformers`, not metadata informers

Reply:
Renamed and replaced, added a comment about why the name changed.  No object count quota unit tests it looks like, may add those in a follow up (although the logic is super simple since it just observes existing object counts, doesn't even crack open the object).